dbms/src/Flash/Executor/PipelineExecutor.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Flash/Executor/PipelineExecutor.h>
+#include <Flash/Pipeline/Pipeline.h>
+#include <Flash/Pipeline/Schedule/Event/Event.h>
+#include <Interpreters/Context.h>
+
+namespace DB
+{
+PipelineExecutor::PipelineExecutor(
+    const ProcessListEntryPtr & process_list_entry_,
+    Context & context_,
+    const Pipelines & pipelines_)
```

Comment:
Maybe it's better to pass `pipelines_` by value and `std::move` to the `this->pipelines`.

Reply:
This way, `PipelineExecutor` will be the only holder of pipelines.

Reply:
updated.